src/algebra/module/equiv.lean

```diff
@@ -652,3 +652,29 @@ rfl
 end add_comm_group
 
 end add_equiv
+
+section linear_map
+
+variables [ring R] [add_comm_group M] [module R M] (T : M →ₗ[R] M)
+
+def linear_equiv.of_invertible [invertible T] : M ≃ₗ[R] M :=
+{ to_fun := λ x, T x,
+  map_add' := by simp only [map_add, eq_self_iff_true, forall_const],
+  map_smul' := by simp only [linear_map.map_smulₛₗ, eq_self_iff_true, forall_const],
+  inv_fun := λ x, ⅟T x,
+  left_inv := λ x, by simp only [← linear_map.mul_apply, inv_of_mul_self, linear_map.one_apply],
+  right_inv := λ x, by simp only [← linear_map.mul_apply, mul_inv_of_self, linear_map.one_apply] }
+
+lemma linear_equiv.coe_of_invertible [invertible T] :
+  ↑(linear_equiv.of_invertible T) = T := by ext; refl
```

Comment:
The proof should be `rfl`

Reply:
should be, but it doesn't work for some reason

Reply:
The version that's `rfl` has ` ⇑` on the LHS. Both are good lemmas

Reply:
I'll just keep it then, because it works well with my other lemmas